src/xkbcomp/scanner.c

```diff
@@ -33,16 +37,18 @@ number(struct scanner *s, int64_t *out, int *out_tok)
     if (is_hex)
         x = strtoll(start, &end, 16);
     else if (is_float) {
-        /* The parser currently just ignores floats, so the cast is
-         * fine - the value doesn't matter. */
-        #ifdef __GNUC__
-        #pragma GCC diagnostic push
-        #pragma GCC diagnostic ignored "-Wbad-function-cast"
-        #endif
-        x = (long long int) strtold(start, &end);
-        #ifdef __GNUC__
-        #pragma GCC diagnostic pop
-        #endif
+        /* NOTE: strtold does not work reliably, because it depends on the
+         * locale for e.g. the decimal separator (e.g. a period or a comma).
+         * Since the parser does not use floats, we do not care of the actual
+         * value: we just need to ensure that the syntax is correct. So just
+         * truncate it. */
+        x = strtoll(start, &end, 10);
```

Comment:
I'm not sure why we bothered with the `strtold` at all. Can just set `x = 0` and be done with it...

Reply:
We still need to ensure the syntax is correct, even if the value is dropped.

Reply:
Right.

The lexer already ensured the input is `[0-9]*\.[0-9]*`.

Does `strtold` allow `.123` and/or `123.` syntax? In any case I think we can get by with a simple validation, basically len > 1 and (depending on answers to questions above) that first/last chars are not `'.'`.

There's also `00053.53` syntax but I think we should allow it -- probably strtold does too?

Reply:
> Does strtold allow .123 and/or 123.

It allows both, but Xorg xkbcomp allow only the second one. While I do not think the first one is conflicting with other XKB syntax, I do think it is bad syntax. So I would avoid supporting if we need a special case.

> There's also 00053.53 syntax but I think we should allow it -- probably strtold does too?

`strtold` does allow it too. This case seems quite legit to me, e.g. for formatting purposes.

Reply:
.123 cannot be parsed simply for now, as `.` is the `DOT` token. So even currently,  we cannot parse it using `strtold` because we simply never get there!

Reply:
I update the MR to accept `123.` syntax (no decimal part).